core/src/common.cpp

```diff
@@ -362,6 +364,264 @@ namespace irods::http
+			auto crv{_jwk.get_jwk_claim("crv").as_string()};
+
+			// Get x coordinate parameter (JWA)
+			auto x{_jwk.get_jwk_claim("x").as_string()};
+
+			// Get y coordinate parameter (JWA)
+			auto y{_jwk.get_jwk_claim("y").as_string()};
+
+			// Add verification algorithm
+			_verifier.allow_algorithm(jwt::algorithm::es256(jwt::helper::create_public_key_from_ec_components(crv, x, y)));
+		}
+	}
+
+	/// Bleh blah
+	///
+	/// \returns A refernce to the provided jwt::verifier, allowing for chaining.
+	auto add_algorithms_to_verifier(jwt::verifier<jwt::default_clock, jwt::traits::nlohmann_json>& _verifier, const jwt::jwks<jwt::traits::nlohmann_json>& _jwks, std::string_view _alg) -> jwt::verifier<jwt::default_clock, jwt::traits::nlohmann_json>& {
+		namespace logging = irods::http::log;
+
+		// Get the JWK the access token was signed with. This is optional.
+		// See RFC 7515 Section 4.1.4
+		try {
+			auto jwk{_jwks.get_jwk(decoded_token.get_key_id())};
+			add_alg_from_jwk(_verifier, jwk, _alg);
+		}
+		catch (const std::runtime_error& e) {
+			// We cannot pick out the specific key used...
+			// Find JWKs that match any of the following:
+			//   - 'alg' (optional) matches exactly
+			//   - 'kty' (required) matches 'alg' family [EC,RSA,oct]
+			//   - 'use' (required if sign & encrypt keys exist) to be 'sig'
+			//
+			// Additionally, symetric algoritms it /seems/ that
+			// "... the octects of the 'client_secret'..." are used for the algorithm...
+			const auto algorithm_family{_alg.substr(0,2)};
+			std::string search_string;
+
+			// RSA family
+			if (algorithm_family == "RS") {
+				search_string = "RSA";
+			}
+			// EC family
+			else if (algorithm_family == "ES") {
+				search_string = "EC";
+			}
+			// Symmetric algo
+			else if (algorithm_family == "HS") {
+				search_string = "oct";
+			}
+
+			// Go through entire key set
+			std::for_each(std::cbegin(_jwks), std::cend(_jwks), [&_verifier, &_alg, &search_string] (const auto& _jwk) -> void {
+				try {
+					if (auto key_type{_jwk.get_jwk_claim("kty").as_string()}; key_type == search_string) {
+						add_alg_from_jwk(_verifier, _jwk, _alg);
+					}
+				}
+				catch (const std::runtime_error& e) {
+					logging::error("{}: Invalid key [{}]", __func__, e.what());
+					return;
+				}
+			});
+		}
+
+		// Allow for chaining
+		return _verifier;
+	}
+	
+	/// Validates an OAuth 2.0 Access Token using
+	///
+	/// See RFC 9068 for more details on JWT for OAuth 2.0
+	/// See RFC 7515 for details on JSON Web Signature (JWS)
+	/// See RFC 7518 for details on JSON Web Algorithms (JWA)
+	///
+	/// \param[in] _jwt A std::string represeting the JWT to verify.
+	///
+	/// \returns The JWT payload if the token can be validated. Otherwise, an empty std::optional is returned
+	///
+	/// \todo Handle encrypted tokens. MUST reject unencrypted tokens if encryption was negotiated at registration.
+	/// \date 2024-10-16 jwt-cpp does not support encrypted tokens as of current.
```

Comment:
Do we like this sort of comment in doxygen?

Reply:
I'm assuming you're asking about the `\todo`. If so, then no.

If the TODO needs to be handled at a later time, open an issue and link it to this comment. That way you'll know the origin of the issue along with the highlighted code block. Include enough information in the issue so that you or another dev is able to understand the situation.

Reply:
How about the comment on the following line:

> 	/// \date 2024-10-16 jwt-cpp does not support encrypted tokens as of current.

Reply:
What's the motivation for including that info?

Reply:
Mostly to not forget the reason for omitting this feature when we eventually get back to it.

Reply:
Let's put it in an issue if it's something we'll need to address later. No need to have it in the code.

Link it here so we know origin of the issue.

Forgot to mention that if having a TODO in the code is needed, that's fine. You just need to apply the rules defined at https://docs.irods.org/4.3.3/developers/contributing/#todo-comments-in-irods-code.

Let's avoid the doxygen `\todo` directive.

Reply:
Removed comment. We'll see if this is a desired feature later on.